File: openapi-generator/java-play-framework/app/apimodels/DocumentFieldTypeNumberValueTypeMetadata.java
package apimodels;

import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * Additional metadata information for a number field type. Only defined when a field type has valueType: ValueType_Number.
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2019-03-01T05:34:42.270Z[GMT]")

@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class DocumentFieldTypeNumberValueTypeMetadata   {
  @JsonProperty("numDecimalPlaces")
  private Long numDecimalPlaces;

  public DocumentFieldTypeNumberValueTypeMetadata numDecimalPlaces(Long numDecimalPlaces) {
    this.numDecimalPlaces = numDecimalPlaces;
    return this;
  }

   /**
   * Number of decimal places that values for this field type can have.
   * @return numDecimalPlaces
  **/
    public Long getNumDecimalPlaces() {
    return numDecimalPlaces;
  }

  public void setNumDecimalPlaces(Long numDecimalPlaces) {
    this.numDecimalPlaces = numDecimalPlaces;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DocumentFieldTypeNumberValueTypeMetadata documentFieldTypeNumberValueTypeMetadata = (DocumentFieldTypeNumberValueTypeMetadata) o;
    return Objects.equals(numDecimalPlaces, documentFieldTypeNumberValueTypeMetadata.numDecimalPlaces);
  }

  @Override
  public int hashCode() {
    return Objects.hash(numDecimalPlaces);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DocumentFieldTypeNumberValueTypeMetadata {\n");
    
    sb.append("    numDecimalPlaces: ").append(toIndentedString(numDecimalPlaces)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: openapi-generator/dart/lib/model/document_field_type_number_value_type_metadata.dart
part of openapi.api;

class DocumentFieldTypeNumberValueTypeMetadata {
  /* Number of decimal places that values for this field type can have. */
  int numDecimalPlaces = null;
  DocumentFieldTypeNumberValueTypeMetadata();

  @override
  String toString() {
    return 'DocumentFieldTypeNumberValueTypeMetadata[numDecimalPlaces=$numDecimalPlaces, ]';
  }

  DocumentFieldTypeNumberValueTypeMetadata.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['numDecimalPlaces'] == null) {
      numDecimalPlaces = null;
    } else {
          numDecimalPlaces = json['numDecimalPlaces'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'numDecimalPlaces': numDecimalPlaces
    };
  }

  static List<DocumentFieldTypeNumberValueTypeMetadata> listFromJson(List<dynamic> json) {
    return json == null ? new List<DocumentFieldTypeNumberValueTypeMetadata>() : json.map((value) => new DocumentFieldTypeNumberValueTypeMetadata.fromJson(value)).toList();
  }

  static Map<String, DocumentFieldTypeNumberValueTypeMetadata> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, DocumentFieldTypeNumberValueTypeMetadata>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new DocumentFieldTypeNumberValueTypeMetadata.fromJson(value));
    }
    return map;
  }
}

